docs/source/reference/tpu.rst

```diff
@@ -0,0 +1,183 @@
+
+      export PYTHONPATH=/home/gcpuser/sky_workdir/models
+
+      python3 mnist_main.py \
+         --tpu=${TPU_NAME} \
+         --model_dir=${MODEL_DIR} \
+         --data_dir=${DATA_DIR} \
+         --train_epochs=10 \
+         --distribution_strategy=tpu \
+         --download
+
+.. note::
+
+   TPU node requires loading data from a GCS bucket, so we add a :code:`file_mounts` to create a new bucket.
+   Check :ref:`SkyPilot Storage <sky-storage>` for more details.
+
+With the above YAML, you should be able to launch the training job with :code:`sky launch`!
+
+.. code-block:: console
+
+   $ sky launch mnist-tpu-node.yaml -c mycluster
+   ...
+   (tpu_app pid=28961) Epoch 9/10
+   (tpu_app pid=28961) 58/58 [==============================] - 1s 19ms/step - loss: 0.1181 - sparse_categorical_accuracy: 0.9646 - val_loss: 0.0921 - val_sparse_categorical_accuracy: 0.9719
+   (tpu_app pid=28961) Epoch 10/10
+   (tpu_app pid=28961) 58/58 [==============================] - 1s 20ms/step - loss: 0.1139 - sparse_categorical_accuracy: 0.9655 - val_loss: 0.0831 - val_sparse_categorical_accuracy: 0.9742
+   ...
+   (tpu_app pid=28961) {'accuracy_top_1': 0.9741753339767456, 'eval_loss': 0.0831054300069809, 'loss': 0.11388632655143738, 'training_accuracy_top_1': 0.9654667377471924}
+
+
+
+TPU VMs
+--------------------------------
+
+To use TPU VMs, user only needs to add :code:`tpu_vm: True` and the desired TPU runtime version in :code:`accelerator_args` shown below:
+
+.. code-block:: yaml
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+
+Note that :code:`instance_type` is no longer needed because TPU VMs is a standalone host VM that physically connects to the TPU device.
+
+Now we show an example of running `mnist training <https://cloud.google.com/tpu/docs/run-calculation-jax#running_jax_code_on_a_tpu_vm>`_ on TPU VM with JAX.
+
+.. code-block:: yaml
+
+   name: mnist-tpu-vm
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: tpu-vm-base
+         tpu_vm: True
+
+   setup: |
+      git clone https://github.com/google/flax.git
+
+      conda activate flax
+      if [ $? -eq 0 ]; then
+         echo 'conda env exists'
+      else
+         conda create -n flax python=3.8 -y
+         conda activate flax
+         # Make sure to install TPU related packages in a conda env to avoid package conflicts.
+         pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+         pip install --upgrade clu
+         pip install -e flax
+      fi
+
+   run: |
+      conda activate flax
+      cd flax/examples/mnist
+      python3 main.py --workdir=/tmp/mnist \
+      --config=configs/default.py \
+      --config.learning_rate=0.05 \
+      --config.num_epochs=10
+
+This time a GCS bucket is not required as the TPU VM is physically linked to the TPU device, which can access data directly.
```

Comment:
```suggestion
A GCS bucket is not required as the TPU VM is physically linked to the TPU device, which can access data directly.
```
(reader may skip to this section directly, so "This time" may be confusing)

Reply:
Fixed!